src/number_theory/pell.lean

```diff
@@ -38,6 +40,35 @@ Pell's equation
 * Connect solutions to the continued fraction expansion of `√d`.
 -/
 
+namespace zsqrtd
+
+variables {d : ℤ}
+
+-- Note: We put these here to avoid having to import `algebra.star.unitary`
+--       into `number_theory.zsqrtd.basic` just for these three results.
+
+/-- An element of `ℤ√d` has norm equal to `1` if and only if it is contained in the submonoid
+of unitary elements. -/
+lemma norm_eq_one_iff_mem_unitary {a : ℤ√d} : a.norm = 1 ↔ a ∈ unitary ℤ√d :=
+begin
+  rw [unitary.mem_iff_self_mul_star, ← norm_eq_mul_conj],
+  norm_cast,
+end
+
+/-- The kernel of the norm map on `ℤ√d` equals the submonoid of unitary elements. -/
+lemma mker_norm_eq_unitary : (@norm_monoid_hom d).mker = unitary ℤ√d :=
+submonoid.ext (λ x, (monoid_hom.mem_mker _).trans norm_eq_one_iff_mem_unitary)
```

Comment:
```suggestion
submonoid.ext (λ x, norm_eq_one_iff_mem_unitary)
```

Reply:
OK. I didn't notice this is a `rfl` lemma.